packages/services/f-feature-management/src/evaluateRestriction.js

```diff
@@ -0,0 +1,220 @@
+import { logger } from './logger';
+
+/**
+ * Checks that the restriction object has at most one operator specified
+ * @param {object} restriction. Throws if not.
+ */
+function verifyRestrictionObject (restriction) {
+    const operatorsSpecified = ['and', 'or', 'eq', 'neq', 'in', 'nin', 'gt', 'gte', 'lt', 'lte']
+    .filter(operator => restriction[operator]);
+
+    if (operatorsSpecified.length > 1) {
+        throw new Error(`${operatorsSpecified.length} operators specified`, operatorsSpecified);
+    }
+}
+
+/**
+ * Utility function to return true if predicate passes for every item in the array.
+ * @param {array} array
+ * @param {function} predicate - map from the array items
+ * @returns Boolean
+ */
+function all (array, predicate) {
+    const passes = array.filter(i => predicate(i));
+
+    return passes.length === array.length;
+}
+
+/**
+ * Utility function to return true if predicate passes for at least one item in the array.
+ * @param {array} array
+ * @param {function} predicate - map from the array items
+ * @returns Boolean
+ */
+function any (array, predicate) {
+    for (let i = 0; i < array.length; i++) {
+        const item = array[i];
+        if (predicate(item)) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Returns the appropriate value from the context, depending on what the restriction is targetting.
+ * @param {object} restriction
+ * @param {object} context
+ * @returns The appropriate value.
+ */
+function getContextValue (restriction, context) {
+    switch (restriction.property) {
+        case 'Country':
+            return context.country;
+        case 'AppVersion':
+            return context.appVersion;
+        default:
+            throw new Error('Unexpected restriction property: ', restriction.property);
+    }
+}
+
+/**
+ * Checks if the property is equal to the checkValue, according to the appropriate rules for the property.
+ * @param {*} checkValue
+ * @param {object} restriction
+ * @param {object} context
+ * @returns Boolean
+ */
+function isEqual (checkValue, restriction, context) {
+    const contextValue = getContextValue(restriction, context);
+
+    switch (restriction.property) {
+        case 'Country':
+            return checkValue.toLowerCase() === contextValue.toLowerCase();
+        case 'AppVersion':
+            return checkValue === contextValue;
+        default:
+            throw new Error('Unexpected restriction property: ', restriction.property);
+    }
+}
+
+/**
+ * Compares two version numbers of format x0.x1...xn, where xi are positive integers.
+ * 1.2.0 > 1.2.
+ * @param {string} valueA
+ * @param {string} valueB
+ * @returns 1 if A is larger, 0 if equal, -1 if B is larger
+ */
+function compareVersionNumbers (valueA, valueB) {
```

Comment:
Are there unit tests for this method?

Reply:
(And others in this file)

Reply:
No. It is somewhat tested by the e2e tests, but no specific unit tests.  If I get time I'll add some in for this but otherwise we can ticket it up?

Reply:
Added